include/rapidjson/document.h

```diff
@@ -613,7 +613,7 @@ class GenericValue {
             kNullFlag, kFalseFlag, kTrueFlag, kObjectFlag, kArrayFlag, kShortStringFlag,
             kNumberAnyFlag
         };
-        RAPIDJSON_ASSERT(type <= kNumberType);
+        RAPIDJSON_ASSERT(type >= kNullFlag && type <= kNumberType);
```

Comment:
`kNullFlag` == 0, so if you'd rather have `type >= 0 && type < kNumberType`, that will work too. Let me know what your preference is.